### PR TITLE
allow specifying a base-url without appending hard coded values

### PR DESCRIPTION
**Pull request by jacobweinstock**
## Description

Currently the osie installer will always append `/misc/osie` to `conf.MirrorBaseURL`. This doesn't allow the runtime/user the ability to control the exactly URL/URI path. This restriction and hard coded values seems arbitrary and unneeded. As `conf.MirrorBaseURL` is used is multiple installers and there is custom logic around when to append to the URI path, introducing another CLI/env seemed required. 

## Why is this needed

Fixes: #

## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [ ] added unit or e2e tests
- [ ] provided instructions on how to upgrade

### CODE REVIEW

File: cmd/boots/main.go [hunk 1]
```diff
@@ -375,6 +377,7 @@ func newCLI(cfg *config, fs *flag.FlagSet) *ffcli.Command {
 	fs.StringVar(&cfg.kubeconfig, "kubeconfig", "", "The Kubernetes config file location. Only applies if DATA_MODEL_VERSION=kubernetes.")
 	fs.StringVar(&cfg.kubeAPI, "kubernetes", "", "The Kubernetes API URL, used for in-cluster client construction. Only applies if DATA_MODEL_VERSION=kubernetes.")
 	fs.StringVar(&cfg.kubeNamespace, "kube-namespace", "", "An optional Kubernetes namespace override to query hardware data from.")
+	fs.StringVar(&cfg.ipxeBaseURL, "ipxe-base-url", "", "The URL base (will be used unmodified) for where the default installer kernel and initrd are hosted.")
```
Comment (chrisdoherty4) [inline cmd/boots/main.go:380]:
```suggestion
	fs.StringVar(&cfg.ipxeBaseURL, "ipxe-base-url", "", "The URL base hosting the default installer kernel and initrd (will be used unmodified).")
```

**Review by chrisdoherty4 — APPROVED** [review]

**Review by chrisdoherty4 — APPROVED** [review]

**Comment by jacobweinstock** [comment]
Hey @mmlb , @nshalman. if/when you have some cycles would you mind checking this out, please? Thanks!

**Comment by displague** [comment]
The PR description should cover the removal of `--osie-path-override`. I'm not aware of any users today, but this functionality is being removed in this PR and that is worth calling out.

**Comment by jacobweinstock** [comment]
will come back to this if needed.

**Pull request closed without merging**